tests/integration/targets/aci_vrf_leak_internal_subnet/tasks/main.yml

```diff
@@ -0,0 +1,183 @@
+
+- name: create third vrf - creation works
+  cisco.aci.aci_vrf:
+    <<: *aci_vrf_present
+    vrf: test3
+  register: vrf_present
+
+- name: create leak internal subnet - check_mode
+  cisco.aci.aci_vrf_leak_internal_subnet: &aci_leak_internal_subnet
+    <<: *aci_info
+    state: present
+    tenant: ansible_test3
+    vrf: test
+    ip: 1.1.1.2
+    leak_internal_subnet: 
+      - vrf: "test2"
+        tenant: "ansible_test3"
+      - vrf: "test3"
+        tenant: "ansible_test3"
+    description: Ansible Test
+  check_mode: true
+  register: leak_sub_check_mode
+
+- name: create leak internal subnet 
+  cisco.aci.aci_vrf_leak_internal_subnet:
+    <<: *aci_leak_internal_subnet
+  register: leak_sub_present
+
+- name: create leak internal subnet - idempotency check
+  cisco.aci.aci_vrf_leak_internal_subnet: 
+    <<: *aci_info
+    state: present
+    tenant: ansible_test3
+    vrf: test
+    ip: 1.1.1.2
+    leak_internal_subnet: 
+      - vrf: "test2"
+        tenant: "ansible_test3"
+      - vrf: "test3"
+        tenant: "ansible_test3"
+    description: Ansible Test
+  register: leak_sub_idempotent
+
+- name: create second leak internal subnet
+  cisco.aci.aci_vrf_leak_internal_subnet: 
+    <<: *aci_info
+    tenant: ansible_test3
+    vrf: test2
+    leak_internal_subnet: 
+      - vrf: "test"
+        tenant: "ansible_test3"
+    description: Ansible Test
+    ip: 1.1.1.2
+  register: leak_sub_present_2
+
+- name: present asserts
+  assert:
+    that:
+      - vrf_present is changed
+      - leak_sub_check_mode is changed
+      - leak_sub_present.sent.leakRoutes.children.0.leakInternalSubnet.children.0.leakTo.attributes.ctxName == 'test2'
+      - leak_sub_present.sent.leakRoutes.children.0.leakInternalSubnet.children.0.leakTo.attributes.tenantName == 'ansible_test3'
+      - leak_sub_present.sent.leakRoutes.children.0.leakInternalSubnet.children.1.leakTo.attributes.ctxName == 'test3'
+      - leak_sub_present.sent.leakRoutes.children.0.leakInternalSubnet.children.1.leakTo.attributes.tenantName == 'ansible_test3'
+      - leak_sub_idempotent is not changed
+      - leak_sub_present_2.sent.leakRoutes.children.0.leakInternalSubnet.children.0.leakTo.attributes.ctxName == 'test'
+      - leak_sub_present_2.sent.leakRoutes.children.0.leakInternalSubnet.children.0.leakTo.attributes.tenantName == 'ansible_test3'
+
+- name: query all
+  cisco.aci.aci_vrf_leak_internal_subnet: &aci_query
+    <<: *aci_tenant_present
+    state: query
+  register: query_all
+
+- name: query vrf
+  cisco.aci.aci_vrf_leak_internal_subnet: 
+    <<: *aci_tenant_present
+    state: query
+    vrf: test
+  register: query
+
+- name: absent case
+  cisco.aci.aci_vrf_leak_internal_subnet: 
+    <<: *aci_info
+    tenant: ansible_test3
+    vrf: test2
+    leak_internal_subnet: 
+      - vrf: "test"
+        tenant: "ansible_test3"
+    description: Ansible Test
+    ip: 1.1.1.2
+    state: absent
+  register: leak_sub_absent
+
+- name: query asserts
```

Comment:
Seeing your asserts it seems that the objects returned on the query are not the correct ones.

I think query_one should return 1 leakInternalSubnet object and not leakRoutes.
I think  query_all should return >1 leakInternalSubnets not the tenant object.